content/v2.0/reference/flux/stdlib/built-in/transformations/window.md

```diff
@@ -110,3 +112,49 @@ from(bucket:"example-bucket")
   |> window(every: 1mo)
   // ...
 ```
+
+## Troubleshooting
+
+### Missing Time Column
+
+The `window()` transformation requires a time column to divide rows into windows.
+The default time column is `_time`.
+A common cause of this error is using an aggregate method and forgetting to create a new time column.
+
+```js
+from(bucket: "telegraf")
+    |> range(start: -5m)
+    |> window(every: 1m)
+    |> mean()
+    |> window(every: inf) // error, no _time column because mean removed it
+```
+
+This can be fixed by using `duplicate()` to create a time column from another row.
+
+
+```js
+from(bucket: "telegraf")
+    |> range(start: -5m)
+    |> window(every: 1m)
+    |> mean()
+    |> duplicate(column: "_stop", as: "_time")
+    |> window(every: inf) // error, no _time column because mean removed it
+```
+
+### Nil Bounds Passed To Window
+
+When `window()` is used with a non-InfluxDB source, you may see the error `nil bounds passed to window; use range to set the window range`.
+The `range()` function must be used before calling `window()` to set the window range.
+
+```js
+import "experimental/array"
+
+array.from(rows: [
+    {_time: 2020-08-25T09:00:00Z, _value: 2.0},
+    {_time: 2020-08-25T09:30:00Z, _value: 3.0},
+    {_time: 2020-08-25T10:00:00Z, _value: 4.0},
+    {_time: 2020-08-25T10:30:00Z, _value: 5.0},
+])
+    |> range(start: 2020-08-25T09:00:00Z, stop: 2020-08-25T11:00:00Z) // required
+    |> window(every: 1h)
+```
```

Comment:
~~~md
**Common cause:** using `window()` with a non-InfluxDB data source.
`window()` requires time bounds set by `range()` and `range()` is only required when querying InfluxDB.

**Solution:** call `range()` before `window()` to set the window range.

```js
import "experimental/array"

array.from(rows: [
    {_time: 2020-08-25T09:00:00Z, _value: 2.0},
    {_time: 2020-08-25T09:30:00Z, _value: 3.0},
    {_time: 2020-08-25T10:00:00Z, _value: 4.0},
    {_time: 2020-08-25T10:30:00Z, _value: 5.0},
])
    |> range(start: 2020-08-25T09:00:00Z, stop: 2020-08-25T11:00:00Z) // required
    |> window(every: 1h)
```
~~~